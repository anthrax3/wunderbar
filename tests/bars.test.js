const { splitNumber, normalBarBuilder } = require('../lib/bars');

describe('splitNumber', () => {
  it('should split float to integer and float parts', () => {
    const value = 1.23;
    const [int, rest] = splitNumber(value);

    expect(int).toBe(1);
    expect(rest).toBe(0.23);
  });
});

describe('normalBarBuilder', () => {
  const testValues = [-1, 0, 0.1, 1.25, 3.34, 5.43, 7.56, 8.75, 9.89, 10];

  it('should return a proper bar for every value', () => {
    expect(testValues.map(normalBarBuilder)).toMatchSnapshot();
  });
});
