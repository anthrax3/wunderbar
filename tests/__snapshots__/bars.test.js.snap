// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalBarBuilder should return a proper bar for every value 1`] = `
Array [
  "",
  "▏",
  "▏",
  "█▎",
  "███▍",
  "█████▌",
  "███████▋",
  "████████▊",
  "██████████",
  "██████████",
]
`;
